tests/test_pipelines.py

```diff
@@ -0,0 +1,115 @@
+import pytest
+import json
+from datetime import datetime
+from StringIO import StringIO
+import subprocess as sp
+from scrapy.exceptions import DropItem
+from pa11ycrawler.pipelines import (
+    DuplicatesPipeline, DropDRFPipeline, Pa11yPipeline
+)
+
+
+def test_duplicates_pipeline():
+    dup_pl = DuplicatesPipeline()
+    spider = object()
+    # first item: no problem
+    item1 = {"url": "google.com"}
+    processed1 = dup_pl.process_item(item1, spider)
+    assert item1 == processed1
+
+    # second item is different, so no problem
+    item2 = {"url": "edx.org"}
+    processed2 = dup_pl.process_item(item2, spider)
+    assert item2 == processed2
+
+    # third is the same as a previous, so raises an exception
+    item3 = {"url": "google.com"}
+    with pytest.raises(DropItem):
+        dup_pl.process_item(item3, spider)
+
+    # fourth is different, so no problem
+    item4 = {"url": "edx.org/foo"}
+    processed4 = dup_pl.process_item(item4, spider)
+    assert item4 == processed4
+
+    # fifth has other, different properties, but the URL is the same
+    item5 = {"url": "edx.org/foo", "page_title": "TitleCase"}
+    with pytest.raises(DropItem):
+        dup_pl.process_item(item5, spider)
+
+
+def test_duplicates_pipeline_next():
+    dup_pl = DuplicatesPipeline()
+    spider = object()
+    item1 = {"url": "https://courses.edx.org/register?next=foo"}
+    item2 = {"url": "https://courses.edx.org/register?next=bar"}
+    dup_pl.process_item(item1, spider)
+    with pytest.raises(DropItem):
+        dup_pl.process_item(item2, spider)
+
+
+def test_drf_pipeline():
+    drf_pl = DropDRFPipeline()
+    spider = object()
+
+    with pytest.raises(DropItem):
+        drf_pl.process_item({"url": "http://courses.edx.org/api/"}, spider)
+
+    # non-api url is fine, though
+    item = {"url": "http://courses.edx.org/course/whatever"}
+    processed = drf_pl.process_item(item, spider)
+    assert item == processed
```

Comment:
This is a regular [`assert` statement](https://docs.python.org/2/reference/simple_stmts.html#grammar-token-assert_stmt) in Python. If you want a customize the failure message, you can do so with a second expression, like this:
```python
assert 1 == 2, "One is not equal to two"
```

However, I've found that test failures are pretty easy to debug with pytest even without custom failure messages, because of the way that pytest can introspect variables and display their output to you in the failing test. [Here's an example of that.](http://docs.pytest.org/en/latest/example/reportingdemo.html)